src/main/kotlin/net/atos/zac/enkelvoudiginformatieobject/EnkelvoudigInformatieObjectLockService.kt

```diff
@@ -0,0 +1,105 @@
+package net.atos.zac.enkelvoudiginformatieobject
+
+import jakarta.enterprise.context.ApplicationScoped
+import jakarta.inject.Inject
+import jakarta.persistence.EntityManager
+import jakarta.persistence.PersistenceContext
+import jakarta.transaction.Transactional
+import jakarta.transaction.Transactional.TxType.REQUIRED
+import jakarta.transaction.Transactional.TxType.SUPPORTS
+import kotlinx.coroutines.FlowPreview
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.flow.retry
+import kotlinx.coroutines.flow.timeout
+import kotlinx.coroutines.runBlocking
+import net.atos.client.zgw.drc.DrcClientService
+import net.atos.client.zgw.zrc.ZRCClientService
+import net.atos.client.zgw.zrc.model.Zaak
+import net.atos.zac.enkelvoudiginformatieobject.model.EnkelvoudigInformatieObjectLock
+import net.atos.zac.util.UriUtil
+import nl.lifely.zac.util.AllOpen
+import nl.lifely.zac.util.NoArgConstructor
+import java.util.Optional
+import java.util.UUID
+import kotlin.time.Duration.Companion.milliseconds
+import kotlin.time.Duration.Companion.seconds
+
+@ApplicationScoped
+@Transactional(SUPPORTS)
+@AllOpen
+@NoArgConstructor
+class EnkelvoudigInformatieObjectLockService @Inject constructor(
+    private val drcClientService: DrcClientService,
+    private val zrcClientService: ZRCClientService
+) {
+    @PersistenceContext(unitName = "ZaakafhandelcomponentPU")
+    private lateinit var entityManager: EntityManager
+
+    @OptIn(FlowPreview::class)
+    fun waitForLockState(enkelvoudiginformatieobjectUUID: UUID, isLocked: Boolean) = flow<Void> {
+        throw EnkelvoudigInformatieObjectLockRetryException(
```

Comment:
When is this exception now thrown exactly? When the retry loops gives up after 2 seconds? In that case the exception message could make that a bit more clear?

Reply:
Retry is invoked on exception. So in `flow` the exception triggers the retry mechanism. 

There are a lot of things to consider though. `timeout` throws an exception and `onEach` delay is not part of the timeout, but executed before emitting a value from `flow`.

Considering the same blocking effect can be achieved by simple loop and no co-routines + flow I think we might as well go this direction. Wdyt?

Reply:
I agree. I think this is a bit overcomplicated indeed. Only thing is we do not want to block the current thread if possible so it all depends on how we wait I guess.

Reply:
ps: why not convert to Kotlin first and then add the retry stuff in a new PR? This makes it hard to review now I think?

Reply:
Agreed. Removed retry for now

Reply:
Agree